cpp/include/Ice/InputStream.h

```diff
@@ -784,13 +784,13 @@ class ICE_API InputStream : public IceInternal::Buffer
      * Reads a short from the stream.
      * @param v The extracted short.
      */
-    void read(Short& v);
+    void read(std::int16_t& v);
```

Comment:
You should update the doc comment as well. I would suggest something like:
> Unmarshals a Slice short into an int16_t.

We use encode/decode" in IceRPC and marshal/unmarshal in Ice, and it's simpler to stick with the existing terminology.